src/interaction/Simulation.cpp

```diff
@@ -0,0 +1,183 @@
+#include "../include/interaction/Simulation.h"
+#include "../include/vrapp/VRVolumeApp.h"
+
+Simulation::Simulation(VRVolumeApp &controller_app, float time) : m_animation_duration(time), m_controller_app(controller_app),
+                                                                  animation_button_label("ANIMATE"), m_animation_state(STOP)
+{
+}
+
+void Simulation::add_simulation_state(SimulationState &simulationState)
+{
+  float time = m_simulation_states.size() * m_animation_duration;
```

Comment:
should this be `simulation_duration`? that seems in line with `m_animation_duration` and is more descriptive

Reply:
I renamed it